src/c_kzg_4844.c

```diff
@@ -1658,7 +1658,8 @@ C_KZG_RET load_trusted_setup(
     out->max_width = 1ULL << max_scale;
 
     /* Allocate all of our arrays */
-    ret = new_fr_array(&out->roots_of_unity, out->max_width);
+    /* We need max_width+1 when expanding */
+    ret = new_fr_array(&out->roots_of_unity, out->max_width + 1);
```

Comment:
Yeah, I forgot that this is a consequence of https://github.com/jtraglia/c-kzg-4844/pull/2/commits/66b358c10801bf3e04517fb4033c2b33e43d495d

Reply:
No worries. I'm actually going to revert back to the old method, using an intermediate array for the expanded roots. I think that would be the better solution long term.